lib/state/ui/actions.ts

```diff
@@ -31,6 +31,14 @@ export const search: A.ActionCreator<A.Search> = (searchQuery: string) => ({
   searchQuery,
 });
 
+export const setSearchFocus: A.ActionCreator<A.SetSearchFocus> = () => ({
+  type: 'SET_SEARCH_FOCUS',
+});
+
+export const unsetSearchFocus: A.ActionCreator<A.UnsetSearchFocus> = () => ({
+  type: 'UNSET_SEARCH_FOCUS',
+});
```

Comment:
Note that we can leave the action untouched but create two action creators. This is typical as a kind of helper to the state being updated.

```ts
export type SetSearchFocus = Action<'SET_SEARCH_FOCUS', { shouldHaveFocus: boolean; }>;
```

```ts
export const focusSearch: A.ActionCreator<A.SetSearchFocus> = () => ({
  type: 'SET_SEARCH_FOCUS',
  shouldHaveFocus: true,
});

export const unfocusSearch: A.ActionCreator<A.SetSearchFocus> = () => ({
  type: 'SET_SEARCH_FOCUS',
  shouldHaveFocus: false,
});
```

```ts
const searchFocused: A.Reducer<boolean> = (state = false, action) => {
  switch (action.type) {
    case 'SET_SEARCH_FOCUS':
      return action.shouldHaveFocus;

    default:
      return state;
  }
}
```

Reply:
> This is probably all a lot to say that this never should have been in state 🙃

Yeah, I was trying to comprehend the point of this thing and why Search calls SetSearchFocus and it's logically nonsense.

Reply:
It's fine to leave it in for now and replicate the `app-state` behavior. _Changes_ can come more easier _after_ the refactor 👍

Reply:
Hmm... I like it better as two, personally.